plugins/woocommerce-blocks/assets/js/blocks/product-collection/utils.tsx

```diff
@@ -199,17 +212,18 @@ export const useSetPreviewState = ( {
 			const isGenericArchiveTemplate =
 				location.type === LocationType.Archive &&
 				location.sourceData?.termId === null;
-			if ( isGenericArchiveTemplate ) {
-				setAttributes( {
-					__privatePreviewState: {
-						isPreview: !! attributes?.query?.inherit,
-						previewMessage: __(
-							'Actual products will vary depending on the page being viewed.',
-							'woocommerce'
-						),
-					},
-				} );
-			}
+
+			setAttributes( {
+				__privatePreviewState: {
+					isPreview: isGenericArchiveTemplate
+						? !! attributes?.query?.inherit
+						: false,
+					previewMessage: __(
+						'Actual products will vary depending on the page being viewed.',
+						'woocommerce'
+					),
+				},
+			} );
```

Comment:
This change is to disable preview when the term ID is resolved later.

Basically if the preview was set to `true` and `isGenericArchiveTemplate` changed to `false`, there was no way for preview to get disabled.

Reply:
Aha, I see. Good catch! Thanks for taking care of this edge case 🚀